File: javascript/14_json.js
// 14 json

const data = { name: "ulas", age: 20, langs: ["js", "ts"] };

const s = JSON.stringify(data, null, 2);
console.log(s);

const parsed = JSON.parse(s);
console.log(parsed.name);

const fs = require("fs");
fs.writeFileSync("data.json", s);
const loaded = JSON.parse(fs.readFileSync("data.json", "utf8"));
console.log(loaded);

File: python/14_json.py
# 14 json

import json

data = {"name": "ulas", "age": 20, "langs": ["python", "go"]}

# to string
s = json.dumps(data, indent=2)
print(s)

# from string
parsed = json.loads(s)
print(parsed["name"])

# to file
with open("data.json", "w") as f:
    json.dump(data, f, indent=2)

# from file
with open("data.json") as f:
    loaded = json.load(f)
    print(loaded)
